Src/Particle/AMReX_ParticleContainerI.H

```diff
@@ -701,87 +722,142 @@ ParticleContainer<NStructReal, NStructInt, NArrayReal, NArrayInt, Allocator>
-                else
-                {
-                    //
-                    // Note that Cell aggregation assumes that p.rdata(0) is mass and
-                    // that all other components should be combined in a mass-weighted
-                    // average.
-                    //
-                    auto agg_map_it = agg_map.find(cell);
-
-                    if (agg_map_it == agg_map.end())
-                    {
-                        //
-                        // Add the particle.
-                        //
-                        SuperParticleType p = src.getSuperParticle(pindex);
-                        p.id() = VirtualParticleID;
-                        agg_map[cell] = p;
-                    }
-                    else
-                    {
-                        AMREX_ASSERT(agg_map_it != agg_map.end());
-                        const SuperParticleType&  pnew       = src.getSuperParticle(pindex);
-                        SuperParticleType&        pold       = agg_map_it->second;
-                        const Real           old_mass   = pold.rdata(0);
-                        const Real           new_mass   = pnew.rdata(0);
-                        const Real           total_mass = old_mass + new_mass;
-                        //
-                        // Set the position to the center of mass.
-                        //
-                        for (int i = 0; i < AMREX_SPACEDIM; i++)
-                        {
-                            pold.pos(i) = static_cast<ParticleReal>((old_mass*pold.pos(i) + new_mass*pnew.pos(i))/total_mass);
-                        }
-                        AMREX_ASSERT(this->Index(pold, level) == cell);
-                        //
-                        // Set the metadata (presumably velocity) to the mass-weighted average.
-                        //
-                        for (int i = 1; i < NStructReal; i++)
-                        {
-                            pold.rdata(i) = static_cast<ParticleReal>((old_mass*pold.rdata(i) + new_mass*pnew.rdata(i))/total_mass);
-                        }
-                        pold.rdata(0) = static_cast<ParticleReal>(total_mass);
-                    }
-                }
+               //Ordering may make this not unique
+               for (int i = 0; i < NArrayReal; ++i)
+                 amrex::Gpu::Atomic::AddNoRet(&partData(iv,AMREX_SPACEDIM+NStructReal+i), partData(iv,AMREX_SPACEDIM)!=0.0 ? static_cast<Real>(0) : static_cast<Real>(p.rdata(NStructReal+i)));
+
+               for (int i = 0; i < AMREX_SPACEDIM; ++i)
+               {
+                   amrex::Gpu::Atomic::AddNoRet(&partData(iv,i), static_cast<Real>((p.rdata(0)*p.pos(i))));
+               }
+               for (int i = 1; i < NStructReal; ++i)
+               {
+                   amrex::Gpu::Atomic::AddNoRet(&partData(iv,AMREX_SPACEDIM+i), static_cast<Real>((p.rdata(0)*p.rdata(i))));
+               }
+               for (int i = 0; i < 1; ++i)
```

Comment:
As we discussed at the meeting today, throughout this PR, could you use brackets, even if the for loop only contains 1 line?

Reply:
Added braces to make if and for loops not 1-line